nf_core/download.py

```diff
@@ -838,81 +847,107 @@ def rectify_raw_container_matches(self, raw_findings):
                 r"[^\"\'](?P<quote>(?<![\\])[\'\"])(?P<param>(?:.(?!(?<![\\])\1))*.?)\1", container_value
             )
 
+            """
+            eliminate known false positives and create plain list out of the tuples returned by the regex above
+            example result:
+            ['https://depot.galaxyproject.org/singularity/scanpy:1.7.2--pyhdfd78af_0', 'biocontainers/scanpy:1.7.2--pyhdfd78af_0']
+            """
+            container_value_defs = [
+                capture for _, capture in container_value_defs[:] if not capture in ["singularity", "apptainer"]
+            ]
+
             """
             For later DSL2 syntax, container_value_defs should contain both, the download URL and the Docker URI.
-            By breaking the loop upon finding a download URL, we avoid duplication
-            (Container is downloaded directly as well as pulled and converted from Docker)
+            For earlier DSL2, both end up in different raw_findings, so a subsequent deduplication is necessary anyway
+            to not pull a container that is already downloaded directly.
+
+            At this point, we just add everything that is either a URL or a Docker URI to cleaned matches.
+            """
+
+            valid_containers = list(filter(either_url_or_docker.match, container_value_defs))
+
+            if valid_containers:
+                cleaned_matches = cleaned_matches + valid_containers
```

Comment:
Just so I understand, what's the difference between `cleaned_matches` and `valid_containers`? Are they obtained with the same regex?